Source/WebCore/platform/graphics/cg/ImageDecoderCG.cpp

```diff
@@ -711,18 +711,13 @@ bool ImageDecoderCG::canDecodeType(const String& mimeType)
 }
 
 #if ENABLE(QUICKLOOK_FULLSCREEN)
-bool ImageDecoderCG::isPanoramic() const
+bool ImageDecoderCG::isMaybePanoramic() const
 {
     auto imageSize = FloatSize(frameSizeAtIndex(0));
     auto aspectRatio = imageSize.aspectRatio();
 
     constexpr float panoramicImageAspectRatioThreshold = 2.0;
-    if (aspectRatio <= panoramicImageAspectRatioThreshold)
-        return false;
-
-    constexpr float panoramicImageMinDimension = 800;
-    constexpr float panoramicImageMaxDimension = 30000;
-    return imageSize.minDimension() > panoramicImageMinDimension && imageSize.maxDimension() < panoramicImageMaxDimension;
+    return aspectRatio > panoramicImageAspectRatioThreshold;
```

Comment:
If the size of the image is small like { 30 x 10 } and there is no `srcset` around this image in the page, will opening QuickLook in this case be okay?

Should not we make the decision to use QuickLook at a higher level where more context about the image is known?

Reply:
Perhaps we can be more refined with the heuristic, but there's little risk to using Quick Look for fullscreen (it now presents an identical visual experience to the user as regular element fullscreen)